### PR TITLE
Foxy devel

### PR DESCRIPTION
**Pull request by amelhassan**
**Porting ros_azure_iothub to ROS2** 

Major differences: 
- Ros_relay and telemetry are not longer supported meaning you can't subscribe to data that is being published to a topic. 
- Dynamic reconfigure has been replaced by parameters client although the device twin configuration is the same. For a parameter to be updated from the device twin, it must be declared as a parameter under a specific node.

### CODE REVIEW

File: dynamic_tutorials/package.xml [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
   <name>dynamic_tutorials</name>
-  <version>1.0.0</version>
-  <description>This sample package demostrates how to set up up Dynamic Reconfigure for a ROS node.</description>
-  <maintainer email="baoxij@microsoft.com">Baoxi Jia</maintainer>
+  <version>0.0.1</version>
+  <description>
+  Minimal node for testing ros_azure_iothub dynamic reconfiguration of parameters
+  </description>
+
+  <maintainer email="amhassa@microsoft.com">Amel Hassan</maintainer>
```
Comment (ooeygui) [inline dynamic_tutorials/package.xml]: We should use aliases like `riotsquad@microsoft.com`.

**Review by ooeygui — APPROVED** [review]

**Comment by ooeygui** [comment]
There is a conflict which prevents merging. Can you update and resolve?

File: roscpp_azure_iothub/CMakeLists.txt [hunk 2]
```diff
@@ -123,23 +130,33 @@ endif()
 #############
 
 install(TARGETS ${PROJECT_NAME}_node
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+  ARCHIVE DESTINATION lib
```
Comment (ooeygui) [inline roscpp_azure_iothub/CMakeLists.txt:133]: Are there colcon equivalent for lib and bin?

Comment (ooeygui) [reply roscpp_azure_iothub/CMakeLists.txt:133]: I noticed that the dynamic reconfigure cmakefile replaces the install functions with "ament_package", is there an equivalent?

**Review by ooeygui — COMMENTED** [review]
one of the cmake files replaces the install functions with `ament_package` should both do the same?

File: roscpp_azure_iothub/CMakeLists.txt (hunk 2 above)
Comment (amelhassan) [reply roscpp_azure_iothub/CMakeLists.txt:133]: The dynamic reconfigure cmakefile still has the install function, but it's placed above `ament_package`

**Comment by ooeygui** [comment]
Good Stuff!!!

**Pull request merged**